src/main/kotlin/alerts/user/persistence.kt

```diff
@@ -47,7 +57,10 @@ class DefaultUserPersistence(private val queries: UserRepo, private val slackUse
       .toList()
 
   override suspend fun insertSlackUser(slackUserId: SlackUserId): User =
-    queries.insertSlackUserId(slackUserId.slackUserId)
-      .let { User(UserId(it.userId), SlackUserId(it.slackUserId)) }
-      .also { slackUsersCounter.inc() }
+    transactionOperator.executeAndAwait {
+      (queries.findBySlackUserId(slackUserId.slackUserId) ?:
+        UserDTO(queries.insertSlackUserId(slackUserId.slackUserId), slackUserId.slackUserId)
+          .also { slackUsersCounter.inc() }
+      ).let { User(UserId(it.userId), SlackUserId(it.slackUserId)) }
```

Comment:
True, sometimes I get dragged by the nice functions of the standard library... forgetting it may also overcomplicate the  code if I chain too much functions 😅

Reply:
It's a common pitfall when learning languages. I feel in a same trap both when learning FP, and when learning Kotlin 😅